### DISCUSSION TITLE
How does excalidraw, selects a drawing

### DISCUSSION THREAD

**Discussion by ptedu3** [excalidraw/excalidraw#2985] [Q&A]
Hey Guys,
Can you help me understand how you implement the selection process for drawings (not shapes)? Are they really drawings on canvas or they are some div layer or something else? My assumption is the first, in which case, how do you select the drawing?

Thanks

**Comment by dwelle** [comment]
Everything you draw is on canvas (not sure what you mean by drawing/shape distinction?). Selection is handled by checking the shape's `x1`/`y1`/`x2`/`y2` (or `points` in case of lines) position against click position.

https://github.com/excalidraw/excalidraw/blob/1801048763857529e189ece2ac2a9300450e47f3/src/components/App.tsx#L2415

(it's kinda a mess in there)
